url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/frontend/handlers.java

```diff
@@ -0,0 +1,959 @@
+            {
+                FrontEnd.filepath = FrontEnd.EditorPane.getToolTipTextAt(id);
+                out = new BufferedWriter(new FileWriter(FrontEnd.filepath));
+                out.write(FrontEnd.activepane.getText());
+            } catch (IOException ex)
+            {
+               // FrontEnd.statuswindow.append("Error occured while saving file...\n");
+                FrontEnd.appendToPane(FrontEnd.statuswindow,"Error Occured While Saving File\n",Color.BLACK);
+                //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+             //   FrontEnd.statuswindow.append(ex.getStackTrace().toString());
+                FrontEnd.appendToPane(FrontEnd.statuswindow,ex.getStackTrace().toString(),Color.BLACK);
+                //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+            } finally
+            {
+                try
+                {
+                    out.close();
+                } catch (IOException ex)
+                {
+                    //FrontEnd.statuswindow.append("Error occured while saving file...\n");
+                    FrontEnd.appendToPane(FrontEnd.statuswindow,"Error Occured While Saving File/n",Color.BLACK);
+                    //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                    //FrontEnd.statuswindow.append(ex.getStackTrace().toString());
+                    FrontEnd.appendToPane(FrontEnd.statuswindow,ex.getMessage(),Color.BLACK);
+                    //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                    //frontend.FrontEnd.exceptionraised++;
+                }
+            }
+
+        }
+      }
+
+    void exit()
+      {
+        System.exit(0);
+      }
+
+    void cut() throws BadLocationException
+      {
+        String sel = FrontEnd.activepane.getSelectedText();
+        frontend.FrontEnd.activepane.getHighlighter().addHighlight(FrontEnd.activepane.getSelectionStart(), FrontEnd.activepane.getSelectionEnd(), frontend.handlers.cyanPainter);
+        StringSelection ss = new StringSelection(sel);
+        FrontEnd.clip.setContents(ss, ss);
+
+        FrontEnd.activepane.replaceSelection("");
+      }
+
+    void copy()
+      {
+        String sel = FrontEnd.activepane.getSelectedText();
+        StringSelection clipString = new StringSelection(sel);
+        FrontEnd.clip.setContents(clipString, clipString);
+      }
+
+    void paste()
+      {
+        Transferable cliptran = FrontEnd.clip.getContents(this);
+        try
+        {
+            String sel = (String) cliptran.getTransferData(DataFlavor.stringFlavor);
+            FrontEnd.activepane.replaceSelection(sel);
+        } catch (Exception ex)
+        {
+            Logger.getLogger(handlers.class.getName()).log(Level.SEVERE, null, ex);
+        }
+
+      }
+
+    void run()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 56 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.